### DISCUSSION TITLE
[golang] Renovate updating golang.org/x/exp to invalid version `aacd6d4b4611`

### DISCUSSION THREAD

**Discussion by stan-stately** [renovatebot/renovate#26342] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.64.3-full

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I dont think this ever worked. I think it broke the first time the package digest was updated

### Describe the problem

Renovate is updating golang.org/x/exp to version `aacd6d4b4611` which is not a valid version. It seems that instead of updating the digest it overwrite the whole version

### Relevant debug logs

<details><summary>Logs</summary>

```

               {
                 "managerData": {"lineNumber": 35, "multiLine": true},
                 "depName": "golang.org/x/exp",
                 "depType": "require",
                 "currentValue": "v0.0.0-20231206192017-f3f8817b8deb",
                 "datasource": "go",
                 "currentDigest": "f3f8817b8deb",
                 "digestOneAndOnly": true,
                 "updates": [
                   {
                     "updateType": "digest",
                     "newValue": "v0.0.0-20231206192017-f3f8817b8deb",
                     "newDigest": "aacd6d4b4611949ff7dcca7a0118e9312168a5f8",
                     "branchName": "renovate-action/all"
                   }
                 ],
                 "packageName": "golang.org/x/exp",
                 "versioning": "semver",
                 "warnings": [],
                 "currentVersion": "v0.0.0-20231206192017-f3f8817b8deb",
                 "fixedVersion": "v0.0.0-20231206192017-f3f8817b8deb"
               }
...
DEBUG: gomod: need to update digest (repository=****, branch=****)
       "depName": "golang.org/x/exp",
       "lineToChange": "\tgolang.org/x/exp v0.0.0-20231206192017-f3f8817b8deb",
       "newDigestRightSized": "aacd6d4b4611"
```

</details>

### Have you created a minimal reproduction repository?

https://github.com/stan-stately/renovate-golang-invalid-version-repro

**Comment by stan-stately** [comment]
It looks like this: https://github.com/renovatebot/renovate/issues/20316

**Comment by rarkins** [comment]
Please install the app into the reproduction repo to demonstrate the incorrect PR

**Comment by stan-stately** [comment, marked as answer]
i just did that an noticed that its fixed in the latest renovate. closing
